internal/db/queryBuilder.go

```diff
@@ -185,3 +208,27 @@ func (qb *queryBuilder) build() (string, []any) {
 
 	return query, args
 }
+
+func (qb *queryBuilder) build() (string, []any) {
+	var query string
+
+	if qb.vrange.From == 0 {
+		query = qb.queryWithoutRemovals()
+	} else {
+		query = qb.queryWithRemovals()
+	}
+
+	query, args := qb.replaceQueryArgs(query)
+
+	return query, args
+}
+
+func AsCte(query string, alias string) string {
+	template := `
+	WITH %s as (
```

Comment:
As mentioned elsewhere, SQL is idented an extra time and `as` is a SQL identifier and should be capitalized.